extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/kernel/controller/ControllerStageTracker.java

```diff
@@ -397,15 +418,11 @@ private void generateResultPartitionsAndBoundaries()
    *
    * @param fault reason why this stage has failed
    */
-  private void failForReason(final MSQFault fault)
+  void failForReason(final MSQFault fault)
   {
     transitionTo(ControllerStagePhase.FAILED);
 
     this.failureReason = fault;
-
-    if (resultKeyStatisticsCollector != null) {
```

Comment:
is there a reason we are reverting this change ?

Reply:
resultKeyStatisticsCollector has been moved from controller stage tracker, there shouldn't be any large memory object to clear here.